src/components/HeadHrefLangs.astro

```diff
@@ -1,11 +1,18 @@
 ---
 import i18next from "i18next";
-import { localizeUrl } from "../..";
+import { localizeUrl, AstroI18next } from "../..";
 
 const supportedLanguages = i18next.languages;
+const defaultLanguage = AstroI18next.config.defaultLocale;
 const currentUrl = Astro.url.href;
 ---
 
+<Fragment>
+  <link
+  rel="alternate"
+  hreflang="x-default"
+  href={localizeUrl(currentUrl, defaultLanguage)}
+  />
```

Comment:
had the same issue but solved it in a different way to keep the render data driven

```astro
---
const supportedLanguages = i18next.languages;
const currentUrl = Astro.url.href;
const defaultLanguage = AstroI18next.config.defaultLocale;

const items = [
    { hrefLang: 'x-default', language: defaultLanguage }, 
    ...supportedLanguages?.map((language) => ({ hrefLang: language, language }))
].filter(Boolean)

---

{
  items?.map(({hrefLang, language}) => (
    <link
      rel="alternate"
      hreflang={hrefLang}
      href={localizeUrl(currentUrl, language)}
    />
  ))
}
```